posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc

```diff
@@ -0,0 +1,145 @@
+blog_description: With Open Liberty 20.0.0.9 you can now try out the MicroProfile GraphQL third-party APIs. This release also includes a built-in user interface for GraphiQL. Plus there has been a number of significant bug fixes
+---
+= Quicker, easier GraphQL queries with Open Liberty 20.0.0.9
+AUTHOR_NAME <https://github.com/jakub-pomykala>
+:imagesdir: /
+:url-prefix:
+:url-about: /about/
+
+// tag::intro[]
+With Open Liberty 20.0.0.9 you can now experiment with the type-safe SmallRye GraphQL Client API, and more easily write and run GraphQL queries and mutations using a built-in GraphiQL user interface. Plus there has been a number of significant bug fixes.
+
+In link:{url-about}[Open Liberty] 20.0.0.9:
+
+* <<GraphQLAPIs, Experiment with a third-party GraphQL client API>>
+* <<GraphiQL, Built-in GraphiQL user interface for faster queries and mutations>>
+* <<bugs, Significant bugs fixed in this release>>
+
+// end::intro[]
+// tag::run[]
+[#run]
+
+== Run your apps using 20.0.0.9
+If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:
+[source,xml]
+----
+<dependency>
+    <groupId>io.openliberty</groupId>
+    <artifactId>openliberty-runtime</artifactId>
+    <version>20.0.0.9</version>
+    <type>zip</type>
+</dependency>
+----
+Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+[source,gradle]
+----
+dependencies {
+    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[20.0.0.9,)'
+}
+----
+Or if you're using Docker:
+[source]
+----
+FROM open-liberty
+----
+//end::run[]
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+//tag::features[]
+[#GraphQLAPIs]
+== Experiment with a third-party GraphQL client API
+
+MicroProfile GraphQL has only been available in Open Liberty for a few months now, and it is already a hit! That said, there are lots of things we'd like to do to improve it - and make it more complete. One of those things is a Client API. While the official Client API is not expected until the next release of MicroProfile GraphQL, you can now experiment with the type-safe SmallRye GraphQL Client API. SmallRye is the underlying implementation of MicroProfile GraphQL, and you can access some of its above-and-beyond-the-spec features by adding the "third-party" API type visibility to your application, like so:
+[source,xml]
+----
+<application name="MyGraphQLApp" location="MyGraphQLApp.war">
+    <classloader apiTypeVisibility="+third-party"/>
+</application>
+----
+Now you can access SmallRye GraphQL APIs like the type-safe client!  Note that these APIs may change in future releases as SmallRye is constantly evolving.  For more information on the SmallRye GraphQL project, please visit https://github.com/smallrye/smallrye-graphql
```

Comment:
```suggestion
Now you can access SmallRye GraphQL APIs like the type-safe client!  Note that these APIs may change in future releases as SmallRye is constantly evolving.  For more information on the SmallRye GraphQL project, please visit https://github.com/smallrye/smallrye-graphql - in Open Liberty 20.0.0.9, we are using SmallRye GraphQL 1.0.7.

```

Updating to include the version of SmallRye that we ship - and adding a line break to (hopefully) separate the two paragraphs.

Reply:
Done